src/qibolab/instruments/emulator/readout.py

```diff
@@ -109,34 +115,28 @@ def simulate_and_demodulate(self, s21: complex, pulse: ReadoutPulse):
         Returns:
             IQ (complex): IQ data for a shot.
         """
-        logmag = np.abs(s21)
-        phase = np.angle(s21)
+        reflected_amplitude = np.abs(s21)
+        reflected_phase = np.angle(s21)
 
-        env_I, env_Q = pulse.envelope_waveforms(self.sampling_rate / 1e9)
+        env_I, env_Q = pulse.envelope_waveforms(
+            self.sampling_rate / 1e9
+        )  # Gigasample per second
 
-        start = int(pulse.start * 1e-9 * self.sampling_rate)
-        t = np.arange(start, start + len(env_I)) / self.sampling_rate
-        reference_signal_I = np.sin(
-            2 * np.pi * pulse.frequency * t + pulse.relative_phase
-        )
-        reference_signal_Q = np.cos(
-            2 * np.pi * pulse.frequency * t + pulse.relative_phase
-        )
+        start = int(pulse.start * 1e-9 * self.sampling_rate)  # n = gigasample index
+        t = np.arange(start, start + len(env_I)) / self.sampling_rate  # t_n
 
-        waveform = (
-            logmag
-            * pulse.amplitude
-            * (
-                env_I.data
-                * np.cos(pulse.relative_phase + phase)
-                * np.sin(2 * np.pi * pulse.frequency * t)
-                + env_Q.data
-                * np.sin(pulse.relative_phase + phase)
-                * np.cos(2 * np.pi * pulse.frequency * t)
-                + self.noise_model(t)
-            )
-        )
+        # Low-pass filtered I-component (with intermediate frequency = carrier frequency)
+        I_filtered = reflected_amplitude * np.cos(
```

Comment:
Try to follow the naming conventions, even when conflicting with the more frequent symbols used in equations
```suggestion
        i_filtered = reflected_amplitude * np.cos(
```